niworkflows/utils/bids.py

```diff
@@ -304,3 +306,42 @@ def _grp_echos(x):
         action = getattr(ses_uids, 'append' if len(bold) > 2 else 'extend')
         action(bold)
     return ses_uids
+
+
+def check_pipeline_version(cvers, data_desc):
+    """
+    Searches for existing BIDS pipeline output and compares against current pipeline version.
+    Returns a warning string if there is a difference between versions, otherwise ``None``.
+
+    .. testsetup::
+
+    >>> import json
+    >>> data = {"PipelineDescription": {"Version": "1.1.1rc5"}}
+    >>> desc_file = Path(tmpdir) / 'sample_dataset_description.json'
+    >>> desc_file.write_text(json.dumps(data))
+    48
+
+    Parameters
+    ----------
+    cvers : :obj:`str`
+        Current pipeline version
+    data_desc : :obj:`str` or :class:`os.Pathlike`
+        Path to pipeline output's dataset_description.json
+
+    Examples
+    --------
+    >>> check_pipeline_version('1.1.1rc5', 'sample_dataset_description.json')
+    >>> check_pipeline_version('1.1.1rc5+129.gbe0e5158', 'sample_dataset_description.json')
```

Comment:
what do you think about dropping the private part of the version in this context, @effigies?

Reply:
Yeah, I think it's probably fine? It should only bite us with people playing with `:unstable`, and they presumably know there are risks.